### DISCUSSION TITLE
Enumerated Column Values : Gender

### DISCUSSION THREAD

**Discussion by emeraldjava** [supabase/supabase#1346] [Questions] [answered]
Hi,
Is it possible to define a varchar column which can only have specific predetermined values. In my case I want to track gender, and acknowledge the fact that this isn't a binary state.
Thanks,
emeraldjava

**Comment by inian** [comment, marked as answer]
Hi @emeraldjava you can do this via a custom type

```
CREATE TYPE gender_type AS ENUM ('male', 'female', 'non_binary');

CREATE TABLE user (
    gender gender_type
);
```

**Reply by emeraldjava** [reply]
thanks - didn't spot the 'TYPE' in the first statement. There's a small type on line 4, gender_tyoe -> gender_type.
